File: core/src/net/brenig/pixelescape/game/player/PlayerPathEntity.java
package net.brenig.pixelescape.game.player;

import net.brenig.pixelescape.game.entity.*;
import net.brenig.pixelescape.game.entity.EntityPlayer;
import net.brenig.pixelescape.lib.Reference;

/**
 * Player path entity
 */
public class PlayerPathEntity implements IMovingEntity {
	private float yVelocity = 0;
	private float yPos = Reference.GAME_RESOLUTION_Y / 2;
	private int xPosScreen = 0;

	public PlayerPathEntity(float yPos, int xPosScreen) {
		this.yPos = yPos;
		this.xPosScreen = xPosScreen;
	}

	public void update(IMovingEntity e, float delta, EntityPlayer playerEntity) {
		this.yPos += yVelocity * delta;
		this.yVelocity = (e.getYPos() - this.getYPos()) * Reference.PATH_ENTITY_ACCELERATION_MOD * playerEntity.getXVelocity();
	}

	public float getYPos() {
		return yPos;
	}

	public int getXPosScreen() {
		return xPosScreen;
	}

	public void reset(float yPos, int xPosScreen) {
		this.yPos = yPos;
		this.yVelocity = 0;
		this.xPosScreen = xPosScreen;
	}

	public void setXPosScreen(int xPosScreen) {
		this.xPosScreen = xPosScreen;
	}

	@SuppressWarnings("SameReturnValue")
	public int getSize() {
		return Reference.PATH_ENTITY_SIZE;
	}

	public int getSizeRadius() {
		return getSize() / 2;
	}
}

File: core/src/net/brenig/pixelescape/game/player/PlayerPathEntity.kt
package net.brenig.pixelescape.game.player

import net.brenig.pixelescape.game.entity.IMovingEntity
import net.brenig.pixelescape.game.entity.impl.EntityPlayer
import net.brenig.pixelescape.game.data.constants.Reference

/**
 * Player path entity
 */
class PlayerPathEntity(yPos: Float, xPosScreen: Int) : IMovingEntity {
    private var yVelocity = 0f
    override var yPos = (Reference.GAME_RESOLUTION_Y / 2).toFloat()
        private set
    var xPosScreen = 0

    val size: Int
        get() = Reference.PATH_ENTITY_SIZE

    val sizeRadius: Int
        get() = size / 2

    init {
        this.yPos = yPos
        this.xPosScreen = xPosScreen
    }

    fun update(e: IMovingEntity, delta: Float, playerEntity: EntityPlayer) {
        this.yPos += yVelocity * delta
        this.yVelocity = (e.yPos - this.yPos) * Reference.PATH_ENTITY_ACCELERATION_MOD * playerEntity.xVelocity
    }

    fun reset(yPos: Float, xPosScreen: Int) {
        this.yPos = yPos
        this.yVelocity = 0f
        this.xPosScreen = xPosScreen
    }
}
